src/media/js/addon/actions/review.js

```diff
@@ -38,10 +50,15 @@ export function publish(slug) {
       urlJoin(process.env.MKT_API_ROOT, `extensions/queue/${slug}/publish/`)
     ).q(apiArgs);
 
+    dispatch(publishPending(slug));
     req
       .post(publishUrl)
       .then((res, err) => {
-        dispatch(publishOk(slug));
+        if (res.status >= 400) {
```

Comment:
Copied what you did here: https://github.com/mozilla/marketplace-content-tools/blob/master/src/media/js/addon/actions/submit.js#L117